services/requester/requester.go

```diff
@@ -248,6 +263,38 @@ func (e *EVM) Call(ctx context.Context, address common.Address, data []byte) ([]
 	return bytesFromCadenceArray(value)
 }
 
+func (e *EVM) EstimateGas(ctx context.Context, data []byte) (uint64, error) {
+	e.logger.Debug().
+		Str("data", fmt.Sprintf("%x", data)).
+		Msg("estimate gas")
+
+	txData := cadenceArrayFromBytes(data).WithType(byteArrayType)
+
+	value, err := e.client.ExecuteScriptAtLatestBlock(
+		ctx,
+		e.replaceAddresses(estimateGasScript),
+		[]cadence.Value{txData},
+	)
+	if err != nil {
+		return 0, fmt.Errorf("failed to execute script: %w", err)
+	}
+
+	// sanity check, should never occur
+	if _, ok := value.(cadence.Array); !ok {
+		e.logger.Panic().Msg(fmt.Sprintf("failed to convert value to array: %v", value))
+	}
+
+	result := value.(cadence.Array)
```

Comment:
can we add a todo here, to have a decoder for EVM.Result. For now this is ok but in future I would want to have unified decoder from EVM.Result back to Go type so we can reuse it.

Reply:
Good point, I think we might need it for the other scripts/transactions that use it.

Reply:
Added TODO in https://github.com/onflow/flow-evm-gateway/pull/81/commits/fac85cb67c8431ed48f45c170d5ecfbd9c1fb2cf